seedemu/services/CAService.py

```diff
@@ -0,0 +1,405 @@
+    from seedemu.services.WebService import WebServer
+    from seedemu.core import Node, Filter
+from seedemu.core import Service, Server
+
+CaFileTemplates: Dict[str, str] = {}
+
+CaFileTemplates["certbot_renew_cron"] = """\
+# /etc/cron.d/certbot: crontab entries for the certbot package
+#
+# Upstream recommends attempting renewal
+#
+# Eventually, this will be an opportunity to validate certificates
+# haven't been revoked, etc.  Renewal will only occur if expiration
+# is within 8 hours.
+#
+# Important Note!  This cronjob will NOT be executed if you are
+# running systemd as your init system.  If you are running systemd,
+# the cronjob.timer function takes precedence over this cronjob.  For
+# more details, see the systemd.timer manpage, or use systemctl show
+# certbot.timer.
+SHELL=/bin/sh
+PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
+
+* */1 * * * root test -x /usr/bin/certbot -a \! -d /run/systemd/system && perl -e 'sleep int(rand(3600))' && REQUESTS_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt certbot -q renew
+"""
+
+def ipsInNetwork(ips: Iterable, network: str) -> bool:
+    """!
+    @brief Check if any of the IPs in the iterable is in the network.
+    This function supports both IPv4 and IPv6 via IPv4-Mapped IPv6 Address.
+
+    @param ips The iterable of IPs.
+
+    @param network The network.
+
+    @returns True if any of the IPs is in the network, False otherwise.
+    """
+    net = ip_network(network)
+    map6to4 = int(IPv6Address('::ffff:0:0'))
+    if isinstance(net, IPv4Network):
+        net = IPv6Network(
+            # convert to IPv4-Mapped IPv6 Address for computation
+            #   ::ffff:V4ADDR
+            # 80 + 16 +  32
+            # https://datatracker.ietf.org/doc/html/rfc4291#section-2.5.5.2
+            f'{IPv6Address(map6to4 | int(net.network_address))}/{96 + net.prefixlen}'
+        )
+    for ip in ips:
+        ip = ip_address(ip)
+        if isinstance(ip, IPv4Address):
+            ip = IPv6Address(map6to4 | int(ip))
+        if ip in net:
+            return True
+    return False
+
+class CAServer(Server):
+    def __init__(self, duration: str, caStore: RootCAStore, step_version: str):
+        super().__init__()
+        self.__duration = duration
+        self.__ca_store = caStore
+        self._step_version = step_version
+
+    def install(self, node: Node):
+        """!
+        @brief Install the CA on the node.
+        """
+        node.addSoftware("ca-certificates")
+        node.addBuildCommand(
+            f"\
+if uname -m | grep x86_64 > /dev/null; then \
+curl -O -L https://github.com/smallstep/certificates/releases/download/v{self._step_version}/step-ca_{self._step_version}_amd64.deb && \
+apt install -y ./step-ca_{self._step_version}_amd64.deb; \
+else \
+curl -O -L https://github.com/smallstep/certificates/releases/download/v{self._step_version}/step-ca_{self._step_version}_arm64.deb && \
+apt install -y ./step-ca_{self._step_version}_arm64.deb; \
+fi"
```

Comment:
How about getting platform information from user as a parameter and install the required software?
```suggestion
            f"\
if uname -m | grep x86_64 > /dev/null; then \
curl -O -L https://github.com/smallstep/certificates/releases/download/v{self._step_version}/step-ca_{self._step_version}_amd64.deb && \
apt install -y ./step-ca_{self._step_version}_amd64.deb; \
else \
curl -O -L https://github.com/smallstep/certificates/releases/download/v{self._step_version}/step-ca_{self._step_version}_arm64.deb && \
apt install -y ./step-ca_{self._step_version}_arm64.deb; \
fi"
```

Reply:
Yes, getting platform information from user is the right way to do that. Although the method in the code works perfectly, making it consistent with the rest of the emulator code is essential. We would like to standardize how the platform information is obtained:  from the user specification.